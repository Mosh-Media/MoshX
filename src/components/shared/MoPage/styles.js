const styles = theme => ({
  section: {
    maxHeight: "100%",
    height: "100%",
    width: "100%",
    paddingTop: theme.padding?.lg,
    paddingBottom: theme.padding?.lg,
    [theme.breakpoints.up(400)]: {
      paddingTop: theme.padding?.lg,
      paddingBottom: theme.padding?.lg
    }
  }
});

export default styles;
