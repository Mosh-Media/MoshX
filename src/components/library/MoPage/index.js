/**
 * MoPage
 * Default page used through out the app, used to wrap elements with page like styles
 * provides a heading title and a subtitle for the page with loading spinner
 * @param {Boolean} isAdmin - based on user role
 * @param {Boolean} isLoading - loading or not
 * @param {String} title - Title for the page
 * @param {String} subtitle - Subtitle for the page
 */

import React from "react";

import withStyles from "@material-ui/core/styles/withStyles";
import MoHint from "components/library/MoHint";
import styles from "./styles";
import MoPageHeader from "../MoPageHeader";
import MoSpinner from "../MoSpinner";
import MoPageContent from "../MoPageContent";

const MoPage = ({
  classes,
  children,
  hint,
  isLoading,
  isAdmin,
  subtitle,
  title
}) => {
  if (isLoading) {
    return <MoSpinner isLoading={isLoading} />;
  }
  return (
    <section className={classes.section}>
      <MoPageHeader text={title} isAdmin={isAdmin} />
      <MoPageContent text={subtitle} />
      <MoHint text={hint} />
      {children}
    </section>
  );
};

export default withStyles(styles)(MoPage);
