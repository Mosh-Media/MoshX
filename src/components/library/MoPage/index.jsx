/**
 * MoPage
 * Default page used through out the app, used to wrap elements with page like styles
 * provides a heading title and a subtitle for the page with loading spinner
 * @param {Boolean} isAdmin - based on user role
 * @param {Boolean} isLoading - loading or not
 * @param {String} title - Title for the page
 * @param {String} subtitle - Subtitle for the page
 */

import React from "react";

import withStyles from "@material-ui/core/styles/withStyles";
import MoHint from "components/library/MoHint";
import styles from "./styles";
import MoSpinner from "../MoSpinner";
import MoTypography from "../MoTypography";

const MoPage = ({ classes, children, hint, isLoading, subtitle, title }) => {
	if (isLoading) {
		return <MoSpinner isLoading={isLoading} />;
	}
	return (
		<section className={classes.section}>
			<MoTypography
				color="greyDark"
				font="breeSerif"
				marginBottom="md"
				text={title}
				variant="h2"
			></MoTypography>
			<MoTypography
				color="greyDark"
				font="openSans"
				marginBottom="md"
				text={subtitle}
				variant="h6"
			></MoTypography>
			<MoHint text={hint} />
			{children}
		</section>
	);
};

export default withStyles(styles)(MoPage);
