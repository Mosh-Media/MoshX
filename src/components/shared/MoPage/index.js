import React from "react";

import styles from "./styles";
import { withRouter } from "react-router-dom";
import withStyles from "@material-ui/core/styles/withStyles";
import { compose } from "recompose";
import MoPageHeader from "../MoPageHeader";
import Spinner from "../Spinner";
import MoPageSubtitle from "../MoPageSubtitle";
import { Grid } from "@material-ui/core";

const MoPage = ({ classes, children, loading, subtitle, title, Component }) => {
  return (
    <Grid container className={classes.section}>
      <Grid item md={6}>
        <MoPageHeader title={title} />
      </Grid>
      <Grid item md={6} className={classes.component}>
        {Component && <Component />}
        {subtitle && <MoPageSubtitle subtitle={subtitle} />}
      </Grid>
      <Spinner loading={loading} color="primary" />
      <section className={classes.content}>{children}</section>
    </Grid>
  );
};

export default compose(withStyles(styles), withRouter)(MoPage);
