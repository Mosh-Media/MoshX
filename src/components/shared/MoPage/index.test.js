import React from "react";

import { BrowserRouter as Router } from "react-router-dom";
import MoPage from "./index.js";
import renderer from "react-test-renderer";

it("Should render MoPage", () => {
  const tree = renderer
    .create(
      <Router>
        <MoPage />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
