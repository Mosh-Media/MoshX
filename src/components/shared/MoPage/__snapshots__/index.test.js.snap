// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoPage 1`] = `
<div
  className="MuiGrid-root MoPage-section-3 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
      style={
        Object {
          "opacity": 0,
          "visibility": "hidden",
        }
      }
    >
      <div
        className="MuiBox-root MuiBox-root-141 MoPageHeader-linkText-108"
      />
    </h2>
  </div>
  <div
    className="MuiGrid-root MoPage-component-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
  />
  <section
    className="MoPage-content-1"
  />
</div>
`;
